addons/binding/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/bravia/BraviaHandler.java

```diff
@@ -0,0 +1,414 @@
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The thing handler for a Sony Bravia device. This is the entry point provides a full two interaction between openhab
+ * and the bravia system.
+ *
+ * @author Tim Roberts
+ * @version $Id: $Id
+ */
+public class BraviaHandler extends BaseThingHandler {
+    // Logger
+    private Logger logger = LoggerFactory.getLogger(BraviaHandler.class);
+
+    /**
+     * The configuration for the system - will be recreated when the configuration changes and will be null when not
+     * online
+     */
+    private BraviaConfig _config;
+
+    /**
+     * The protocol handler being used - will be null if not initialized
+     */
+    private BraviaProtocol _protocolHandler;
+
+    /**
+     * The retry connection event - will only be created when we are retrying the connection attempt
+     */
+    private ScheduledFuture<?> _retryConnection;
+
+    /**
+     * The refresh state event - will only be created when we are connected
+     */
+    private ScheduledFuture<?> _refreshState;
+
+    /**
+     * The ping event - will be non-null when online (null otherwise)
+     */
+    private ScheduledFuture<?> _ping;
+
+    /**
+     * Constructs the handler from the {@link Thing}
+     *
+     * @param thing a non-null {@link Thing} the handler is for
+     */
+    public BraviaHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Handles commands to specific channels. This implementation will offload much of its work to the
+     * {@link BraviaProtocol}. Basically we validate the type of command for the channel then call the
+     * {@link BraviaProtocol} to handle the actual protocol. Special use case is the {@link RefreshType}
+     * where we call {{@link #handleRefresh(String)} to handle a refresh of the specific channel (which in turn calls
+     * {@link BraviaProtocol} to handle the actual refresh
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (command instanceof RefreshType) {
+            handleRefresh(channelUID.getId());
+            return;
+        }
+
+        String id = channelUID.getId();
+
+        if (id == null) {
+            logger.warn("Called with a null channel id - ignoring");
+            return;
+        }
+
+        if (id.equals(BraviaConstants.CHANNEL_IR)) {
+            if (command instanceof DecimalType) {
+                _protocolHandler.setIR(((DecimalType) command).intValue());
+            } else {
+                logger.error("Received a IR channel command with a non StringType: " + command);
```

Comment:
warn should be enough here
please use parameterized logging (`"non StringType: {}", command`).